app/controllers/api/stories_controller.rb

```diff
@@ -11,16 +11,26 @@ def random
     show
   end
 
-  def resource
-    @story ||= if params[:id]
-                 Story.published.visible.find(params[:id])
-               elsif request.put? || request.post?
-                 Story.new
-               end
+  private
+
+  def resources_base
+    if highlighted?
+      account.portfolio_stories
+    else
+      super
+    end
   end
 
-  def resources
-    @stories ||= resources_base.includes(
+  def filtered(resources)
+    if highlighted?
+      resources
+    else
+      super
+    end
+  end
+
+  def included(relation)
```

Comment:
Is this going to be used for the `#show` cache key method? I think previously I tried not to do that.
Absolutely great that `#included` is excluded for the `index` cache key, but should it also be excluded for `show`?